### PR TITLE
Added FlySky IBUS support by Cesco/Plüschi

### PR DESCRIPTION
**Pull request by fiendie**
http://www.multiwii.com/forum/viewtopic.php?f=23&t=1947&p=61095#p61093

### CODE REVIEW

File: src/ibus.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+// Receive ISR callback
+static void ibusDataReceive(uint16_t c)
+{
+    uint32_t ibusTime;
+    static uint32_t ibusTimeLast;
+    static uint8_t ibusFramePosition;
+
+    ibusTime = micros();
+    if ((ibusTime - ibusTimeLast) > 3000)  ibusFramePosition = 0;
```
Comment (trollcop) [inline src/ibus.c]: indentation

File: src/ibus.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    ibusTimeLast = ibusTime;
+
+    if (ibusFramePosition == 0 && c != IBUS_SYNCBYTE) return;
```
Comment (trollcop) [inline src/ibus.c]: indentation

File: src/ibus.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		
+		ibus[ibusFramePosition] = (uint8_t)c;
```
Comment (trollcop) [inline src/ibus.c]: spacing and/or wrong indentation

File: src/ibus.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    if (ibusFramePosition == IBUS_BUFFSIZE - 1) {
+        ibusFrameDone = true;
+				failsafeCnt = 0;   // clear FailSafe counter
```
Comment (trollcop) [inline src/ibus.c]: utterly fail indentation

File: src/ibus.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+					return true;
+				}			
+    }
+    return false;
+}
+
+static uint16_t ibusReadRawRC(uint8_t chan)
+{
+    return ibusChannelData[mcfg.rcmap[chan]];
+}
```
Comment (trollcop) [inline src/ibus.c]: this entire file needs help

**Comment by fiendie** [comment]
I'll get on that later ;)

**Comment by fiendie** [comment]
@trollcop That should be it

File: src/ibus.c [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+{
+    uint8_t i;
+    uint16_t chksum,rxsum;
+
+    if (ibusFrameDone) {
+        ibusFrameDone = false;
+            
+        chksum = 0xFFFF;
+        
+        for (i=0;i<30;i++) {
```
Comment (trollcop) [inline src/ibus.c]: no need {} on single line for statements

Comment (trollcop) [inline src/ibus.c]: also spacing i = 0 etc

**Pull request merged**